### ISSUE TITLE
about training 4M dataset and the loss converge slowly

### ISSUE THREAD

**Issue by Darylmeng** [xinyu1205/recognize-anything#178]
I collected the 4M datasets with some imgs urls failed (about 1.5%) , which causes 146 tags missing. I regenerate the label embedding, ram_tag_list_en(cn)/threshold according to the missing tags (get 4439 tags in total) and retrain the model. however, the loss converge slowly and the loss_tag/align/diss increase rapidly in some iterations during the first epoch. During the training process, the loss_tag and loss-align are oscillating, but overall they are decreasing. What's more, i found that the loss_diss drop to about 0.28 at the first epoch and almost stop decreasing even the training finished.

I only adjust the hyper-parameter batch_size from 52 to 26 as the reason of OOM(40g A100), while it should not be the key-point causing my problem. I evaluate on Openimages v6 common with my training ckpt and find the results is bad. Can you please offer some suggestions about my question or release the training log or the 4M ckpt for reference. Thank you very much!

pic of  loss increase rapidly and loss-align is oscillating
<img width="1283" alt="image" src="https://github.com/xinyu1205/recognize-anything/assets/51185961/c3ba3700-e0dc-4624-96dc-903feaf17e2e">

pic of average loss at epoch 5
<img width="1101" alt="image" src="https://github.com/xinyu1205/recognize-anything/assets/51185961/6cc1642d-0707-4ffa-9edc-4d31549dc80b">

pic of evaluation on Openimages v6 common with my training ckpt
<img width="174" alt="image" src="https://github.com/xinyu1205/recognize-anything/assets/51185961/b294eed3-1401-45ef-bcdf-375713781eb9">

**Comment by Darylmeng** [comment]
I check the provided 4M data json files and find that only 4441 tags exists, which is smaller than the 4585 tags as claimed in the paper. May be the other 144 tags are in the 12M dataset? Will the missing labels cause bad effects on the training of 4M dataset training? As mentioned above, i relabel the json file and the model can't perform well. 
I would be truly grateful if the author could offer any insights or responses to the questions I've presented.

**Comment by Darylmeng** [comment]
I also raise two more questions. In Appendix A in paper, batch_size is 720. However, the batch_size in config file "pretrain" is set to 52. According to the paper, the model is trained by 8 A100, so the batch_size should be 52X8=416? Also, the loss is computed as the sum of (loss_tag+loss_diss+loss_align). However, i observe the magnitudes of the three losses are in the hundreds, tens, and less than one, respectively. I'm curious if simply adding losses with such significant differences in magnitude without balancing them with any weights will affect the training? Sincerely awaiting for your reply.

**Comment by xinyu1205** [comment]
Did you load the ImageNet-pretrained image encoder weights? It is fixed by subsequent PR.
![image](https://github.com/xinyu1205/recognize-anything/assets/40064457/5583f465-a98f-4dd1-bdd3-70069372a2b6)
